lib/Yam/agama/agama_base.pm

```diff
@@ -35,4 +36,9 @@ sub test_flags {
     return {fatal => 1};
 }
 
+sub upload_traces {
+    assert_script_run("tar czf /tmp/traces.tar.gz test-results/");
```

Comment:
Inside the _ISO_ it seems that `bzip2` is not present: `bzip2: command not found`.
The _sub_ `tar_and_upload_log()` use it to compress the tarball by using the options `tar -jcv -f`.

Modified the _sub_ `upload_traces()`, so that continue in case of failure.

Reply:
we use for yast logs `get_available_compression()`, it happens in minimal system the  same thing. It is quite hard to map a linux command with a wrapper, you never get exactly what you want if you just map its options "a little"like in this case, and if you map it fully, probably doesn't work, and should be done with some cpan module, so it is kind doing bash in Perl. 

We could update that function passing in $args the compression, but that would be just a letter "j" or "z" and that would not be very readable, as there are not enums in perl, we would need constants or passing the whole word (which something there isn't depending on the command arg of a given command) but you can add a comment in the documentation of the function describing both formats and that would help, it would be something like $args->{compression_options} which would take a default value of "j" to not break existing code.
what solution you prefer? we can go both ways.